mm2src/coins/lightning.rs

```diff
@@ -288,8 +489,17 @@ impl SwapOps for LightningCoin {
         _amount: BigDecimal,
         _swap_contract_address: &Option<BytesJson>,
         _swap_unique_data: &[u8],
+        payment_instructions: &Option<PaymentInstructions>,
     ) -> TransactionFut {
-        unimplemented!()
+        let PaymentInstructions::Lightning(invoice) = try_tx_fus!(payment_instructions
+            .clone()
+            .ok_or_else(|| TransactionErr::Plain("payment_instructions can't be None".into())));
+        let coin = self.clone();
+        let fut = async move {
+            let payment = try_tx_s!(coin.pay_invoice(invoice).await);
```

Comment:
`How about using .validate_instructions() at send_maker_payment and send_taker_payment methods instead of in maker_swap.rs and taker_swap.rs?`

I wanted the swap to fail quickly if the instructions are invalid. let's say the maker sends an invalid instructions, if validation is moved right before or inside `send_taker_payment ` then the taker will have to wait for the confirmations of the maker payment first before failing the swap due to invalid instructions. I think the instructions should be checked if valid or not as soon as it's received in the p2p message as currently implemented :)